sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -111,10 +100,24 @@ internal void ApplyRequestContext(RequestContext? context, ResponseClassifier? c
             {
                 ResponseClassifier = context.Apply(classifier);
             }
+
+            Apply(context);
         }
 
         internal List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)>? Policies { get; set; }
 
+        internal static HttpMessage AssertHttpMessage(PipelineMessage message)
```

Comment:
I like the Assert - it's essentially a downcast - we are asserting that the PipelineMessage has a runtime type of HttpMessage, and we will throw if it's not.  If there's established precedent for naming such a method elsewhere, I'm happy to follow, but I think conversion may not be technically correct here?

Reply:
I assumed the intent was convert because we cast and return the casted instance. Assert, to me, implies that we only want to check, but not alter. I realize it's internal and doesn't matter too much, but when seeing it used around the rest of the code, this seemed like an odd name to me considering how it's used.

Reply:
NUnit does it? https://docs.nunit.org/articles/nunit/writing-tests/assertions/classic-assertions/Assert.Throws.html

Reply:
Addressed in https://github.com/Azure/azure-sdk-for-net/pull/42262